### DISCUSSION TITLE
Medieval symbols not rendering in OSMD

### DISCUSSION THREAD

**Discussion by bdrivera** [opensheetmusicdisplay/opensheetmusicdisplay#1529] [Q&A] [answered]
Hello,
I have a bit of a weird question. Long story short, as part of my undergrad capstone project my team has been tasked to create an application to assist the Music school in studying 13th century musical manuscripts and western transcriptions of those pieces. We've decided to use OSMD for this, as nothing else seems to come close to the capability OSMD has.
The client we're developing for currently uses Musescore to create MusicXML, and to represent the notes of the manuscript they use symbols in musescore.

![Musescore_Example](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/54782106/0a89e36d-9ea7-477e-8f73-6e75cfbab303)
It appears that the XML representation of the first symbol would be:
`<direction placement="below">
        <direction-type>
          <other-direction smufl="chantCclef" relative-x="-48.67" relative-y="-20.28"/>
          </direction-type>
        </direction>`

OSMD doesn't visibly render these, and instead puts rests. I'm wondering if these symbols fall under the 'drawHiddenNotes' option. I understand that if this is the case, it's currently unsupported according to the OSMDOptions documentation. I would appreciate any help in understanding how OSMD is interpreting these symbols, and if there is any current or planned option to make them work. My team and I have no musical education beyond what we've learned for this project, so I apologize if this post misuses terminology.

**Comment by sschmidTU** [comment, marked as answer]
@bdrivera Hi, can you share the MusicXML file for this? (you can rename it to .txt or zip it and drop it in a comment here)
The XML excerpt (`<direction>`) doesn't look like it's the symbols in the image, as a direction is usually outside the stafflines (as the `"placement='below'"` indicates).

Here's basically all the glyphs Vexflow (1.2.93 which we're using) has:
https://groovemechanic.net/public/vexflow/src/glyphs.html

If the glyph you want is not in there (which I'm guessing it's not), you would need to add it to our VexFlowPatch (`src/VexFlowPatch/src/fonts/vexflow_font.js`).
You also need to add some code.
An example where we added new glyphs is here: PR https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/pull/1463

by the way, if you change files in VexFlowPatch, you will at least need to run `npm run prebuild` and restart `npm start`, or directly run `npm run build` instead to apply the patch.

It might be that Vexflow 5 has these glyphs now, as it uses Smufl fonts, which we don't yet use (as we still use Vexflow 1.2.93 because of formatting issues and the effort to switch to 5 would be quite big, though it's planned for the future).

**Reply by bdrivera** [reply]
[OSMD_Disc_1529.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/14749333/OSMD_Disc_1529.zip)
I've zipped the xml, along with the Musescore .mscz file just in case.
It doesn't appear the symbols our client is using are in there, so it seems that may be our issue. I appreciate the help!

**Reply by sschmidTU** [reply]
Interesting.
They should probably be written as noteheads by the way, like this:
![image](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/33069673/09cf403a-676f-450f-9701-306a063fd611)
I'll take a look at your file.

**Reply by sschmidTU** [reply]
Oh, I see that some of these symbols are clefs, you'll probably have your work cut out for you adding all of these "obsolete" medieval symbols...
Also, my brand new MuseScore 4 install crashes while opening the MusicXML.
(though the .mscz works fine in 4, doesn't load in 3)

**Reply by bdrivera** [reply]
Thank you for all the help!
I also noticed the MuseScore crashing issue with the MusicXML, not sure what's going on there.
